TribalWars/src/WebApi/Hubs/VillageMaterialsHub.cs

```diff
@@ -0,0 +1,61 @@
+﻿using ApplicationCore.Entities;
+using ApplicationCore.Interfaces.Services;
+using ApplicationCore.Resources;
+using ApplicationCore.ViewModels;
+using AutoMapper;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.SignalR;
+using Microsoft.Extensions.Localization;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace WebApi.Hubs
+{
+    public class VillageMaterialsHub : Hub<IVillageMaterialsClient>
```

Comment:
I think most of this actions should be authorized. You can put [Authorize] annotation on top of the Hub class, if all methods are supposed to be protected.